pkg/test/ginkgo/external.go

```diff
@@ -0,0 +1,131 @@
+package ginkgo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/origin/test/extended/util"
+)
+
+type serializedTest struct {
+	Name   string
+	Labels string
+}
+
+// externalTestsForSuite reads tests from external binary, currently only
+// k8s-tests is supported
+func externalTestsForSuite(ctx context.Context) ([]*testCase, error) {
```

Comment:
Nit: this adds a few seconds to the runtime

Reply:
Correct, up to 10s from what I was testing. But that's also the reason why we have that envvar to opt-out of it.